lib/CL/clCreateContext.c

```diff
@@ -150,20 +171,23 @@ POname(clCreateContext)(const cl_context_properties * properties,
 
   POCL_GOTO_ERROR_COND((pfn_notify == NULL && user_data != NULL), CL_INVALID_VALUE);
 
-  errcode = pocl_init_devices();
-  /* clCreateContext cannot return CL_DEVICE_NOT_FOUND, which is what
-   * pocl_init_devices() returns if no devices could be probed. Hence,
-   * remap this error to CL_INVALID_DEVICE. Note that this particular
-   * situation should never arise, since an application should issue
-   * clGetDeviceIDs before clCreateContext, and we would have returned
-   * CL_DEVICE_NOT_FOUND already from clGetDeviceIDs. Still, no reason
-   * not to handle it.
-   */
-
-  if (errcode == CL_DEVICE_NOT_FOUND)
-    errcode = CL_INVALID_DEVICE;
-  POCL_GOTO_ERROR_ON ((errcode != CL_SUCCESS), errcode,
-                      "Could not initialize devices\n");
+  if (!platform->instance)
```

Comment:
I prefer to be explicit with the predicates: "platform->instance != NULL" instead of relying on the implicit boolean conversion.